imports/plugins/core/ui-navbar/client/components/navbarCustomer.js

```diff
@@ -0,0 +1,187 @@
+    shop: PropTypes.object,
+    visibility: PropTypes.object.isRequired
+  };
+
+  static defaultProps = {
+    visibility: {
+      hamburger: true,
+      brand: true,
+      tags: true,
+      search: true,
+      notifications: true,
+      languages: true,
+      currency: true,
+      mainDropdown: true,
+      cartContainer: true
+    }
+  };
+
+  state = {
+    navBarVisible: false,
+    searchModalOpen: false
+  }
+
+  toggleNavbarVisibility = () => {
+    const isVisible = this.state.navBarVisible;
+    this.setState({ navBarVisible: !isVisible });
+  }
+
+  handleCloseNavbar = () => {
+    this.setState({ navBarVisible: false });
+  }
+
+  handleOpenSearchModal = () => {
+    this.setState({ searchModalOpen: true });
+  }
+
+  handleCloseSearchModal = () => {
+    this.setState({ searchModalOpen: false });
+  }
+
+  renderLanguage() {
+    return (
+      <div className="languages">
+        <Components.LanguageDropdownCustomer />
+      </div>
+    );
+  }
+
+  renderCurrency() {
+    return (
+      <div className="currencies">
+        <Components.CurrencyDropdownCustomer />
+      </div>
+    );
+  }
+
+  renderBrand() {
+    const { brandMedia, shop } = this.props;
+
+    const { name } = shop || {};
+    const logo = brandMedia && brandMedia.url({ store: "large" });
+
+    return (
+      <Components.Brand
+        logo={logo}
+        title={name || ""}
+      />
+    );
+  }
+
+  renderSearchButton() {
+    if (this.props.searchEnabled) {
+      return (
+        <div className="search">
+          <Components.FlatButton
+            icon="fa fa-search"
+            kind="flat"
+            onClick={this.handleOpenSearchModal}
+          />
+          <Components.SearchSubscription
+            open={this.state.searchModalOpen}
+            onClose={this.handleCloseSearchModal}
+          />
+        </div>
+      );
+    }
+  }
+
+  renderNotificationIcon() {
+    if (this.props.hasProperPermission) {
+      return (
+        <div className="navbar-notification">
+          <Components.Notification />
+        </div>
+      );
+    }
+  }
+
+  renderCartContainerAndPanel() {
+    return (
+      <div className="cart-container">
+        <div className="cart">
+          <Components.CartIconCustomer />
+        </div>
+        <div className="cart-alert">
+          <Components.CartPanelCustomer />
+        </div>
+      </div>
+    );
+  }
+
+  renderMainDropdown() {
+    return (
+      <Components.MainDropdownCustomer />
+    );
+  }
+
+  renderHamburgerButton() {
+    return (
+      <div className="showmenu"><Components.Button icon="bars" onClick={this.toggleNavbarVisibility} /></div>
+    );
+  }
+
+  renderTagNav() {
+    return (
+      <header className="menu" role="banner">
+        <Components.TagNavCustomer
+          isVisible={this.state.navBarVisible}
+          closeNavbar={this.handleCloseNavbar}
+          {...this.props}
+        >
+          {/* <Components.Brand />
```

Comment:
Just a some commented code that can be removed, and the `Components.TagNavCustomer` can be self-closing